JS/search.js

```diff
@@ -0,0 +1,49 @@
+import { createPosts } from "./createposts.js";
+
+const userData = "https://api.noroff.dev/api/v1/social/posts?_author=true"
+
+const searchInput = document.getElementById('search-input');
+const postContainer = document.getElementById('postContainer');
+
+
+
+let posts = [];
+
+searchInput.addEventListener("input", (e) => {
+    const searchText = e.target.value.toLowerCase();
+    const filteredPosts = posts.filter(post => {
+        const authorName = post.author.name.toLowerCase();
+        const postTitle = post.title.toLowerCase();
+        return authorName.includes(searchText) || postTitle.includes(searchText);
```

Comment:
What happens when the search returns nothing? As it is, it shows nothing when no patch matches. You do some like this 

```
const searchResult =  authorName.includes(searchText) || postTitle.includes(searchText);
if (searchREsult) {
return searchResult;
} else {
return 'sorry, no record is found"
}```

Reply:
updated, if on results an message will be displayed informing the user